File: Backend/OhDeerBackend/OhDeerBackend/Models/Purchase.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using OhDeerBackend.Models.Base_Classes;

namespace OhDeerBackend.Models
{
    public class Purchase : BaseEntity
    {
        [Required]
        public int Amount { get; set; }

        [Required]
        public TimeSpan Timestamp { get; set; }

        [Required]
        public DateTime Date { get; set; }

        public bool IsOutstanding { get; set; }

        public int RepairRating { get; set; }
        public int GeneralRating { get; set; }

        public string? RatingComment { get; set; }

        // Foreign Keys
        [Required]
        public int PostId { get; set; }
        public virtual Post Post { get; set; }

        [Required]
        public int UserId { get; set; }
        public virtual User User { get; set; }

        [Required]
        public int FeeId { get; set; }
        public virtual Fee Fee { get; set; }
    }

}

File: Backend/app/Models/Purchase.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Purchase extends Model
{
    use HasFactory;

    protected $fillable = [
        'amount',
        'is_outstanding',
        'repair_rating',
        'general_rating',
        'rating_comment',
        'post_id',
        'user_id',
        'fee_id',
    ];

    protected $casts = [
        'is_outstanding' => 'boolean',
    ];

    public function post()
    {
        return $this->belongsTo(Post::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function fee()
    {
        return $this->belongsTo(Fee::class);
    }
}
